### ISSUE TITLE
Debug Test can't receive `DYLD_LIBRARY_PATH` variable in macOS.

### ISSUE THREAD

**Issue by zhen8838** [OmniSharp/omnisharp-vscode#5131]
## Issue Description ##

I set the `DYLD_LIBRARY_PATH` in the zshrc, then run dotnet test print the env variable value. but when I use debug Test , It can not get the variable value.

I record a video:
1. I have not set the `CUSTOM_VAR` in the zshrc, then run the dotnet test, It's can print the `DYLD_LIBRARY_PATH` variable value.
2. I set the `CUSTOM_VAR` in the zshrc, then restart the vscode and click `Debut Test`, Now in terminal we can see the `CUSTOM_VAR` variable, but `DYLD_LIBRARY_PATH ` is null.
3. use dotnet test, I can get both environment variable value.

https://user-images.githubusercontent.com/26156999/160542063-cf23a55a-9de8-4837-ab27-434286b7c298.mov

## Steps to Reproduce ##

1. write the `export DYLD_LIBRARY_PATH="xxx"` in the `~/.zshrc`
4. restart the viscose
5. click the debug test with code:
```csharp
   [Fact]
    public void TestGETENV()
    {
        // var env = System.Environment.GetEnvironmentVariables();
        // foreach (var k in env.Keys)
        // {
        //     System.Console.WriteLine(k + ": " + env[k]);
        // }
        System.Console.WriteLine("Get DYLD_LIBRARY_PATH : " + System.Environment.GetEnvironmentVariable("DYLD_LIBRARY_PATH"));
        System.Console.WriteLine("Get CUSTOM_VAR : " + System.Environment.GetEnvironmentVariable("CUSTOM_VAR"));
    }
```

## Expected Behavior ##

will print the `Get DYLD_LIBRARY_PATH : xxx`

## Actual Behavior ##

`Get DYLD_LIBRARY_PATH : `

### OmniSharp log ###
<details>Post the output from Output-->
[info]: OmniSharp.DotNetTest.VSTestManager
        read: {"MessageType":"TestSession.Connected","Payload":null}
[info]: OmniSharp.DotNetTest.VSTestManager
        read: {"MessageType":"ProtocolVersion","Payload":1}
[info]: OmniSharp.DotNetTest.DebugSessionManager
        Debug session started.
[info]: OmniSharp.DotNetTest.DebugSessionManager
        Debug session ended.
</details>

### C# log ###
<details>Post the output from Output-->C# here</details>

## Environment information ##

**VSCode version**: 1.65.2
**C# Extension**: 1.24.1

<details><summary>Mono Information</summary>
        OmniSharp using built-in mono</details>
<details><summary>Dotnet Information</summary>
.NET SDK (reflecting any global.json):
 Version:   6.0.201
 Commit:    ef40e6aa06

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.3
 OS Platform: Darwin
 RID:         osx.12-arm64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.201/

Host (useful for support):
  Version: 6.0.3
  Commit:  c24d9a9c91

.NET SDKs installed:
  6.0.201 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.3 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.3 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|better-comments|aaron-bond|2.1.0|
|change-case|wmaurer|1.0.0|
|clang-format|xaver|1.9.0|
|codestream|CodeStream|12.13.0|
|copilot|GitHub|1.9.5350|
|csharp|ms-dotnettools|1.24.1|
|doxdocgen|cschlosser|1.4.0|
|file-icons|file-icons|1.0.29|
|gitlens|eamodio|12.0.5|
|hexeditor|ms-vscode|1.9.5|
|icons-carbon|antfu|0.2.4|
|increment-selection|albymor|0.2.0|
|material-theme|zhuangtongfa|3.13.20|
|remote-ssh|ms-vscode-remote|0.76.1|
|remote-ssh-edit|ms-vscode-remote|0.76.1|
|test-adapter-converter|ms-vscode|0.1.5|
|todo-tree|Gruntfuggly|0.0.215|
|vscode-clangd|llvm-vs-code-extensions|0.1.15|
|vscode-lldb|vadimcn|1.7.0|
|vscode-test-explorer|hbenl|2.21.1|
|vscode-wakatime|WakaTime|18.0.8|
|vscode-yaml|redhat|1.5.1|
|vscodeintellicode|VisualStudioExptTeam|1.2.17|;

</details>

**Comment by zhen8838** [comment]
@nohwnd Thank you for your patient reply, but I encountered a problem in the process of `Debug Test`:
https://user-images.githubusercontent.com/26156999/160869585-cb300c0b-5de3-47c8-998b-a618d07366b5.mov

Maybe when debug Test,It's not accept the `.runsettings` file?

**Comment by nohwnd** [comment]
You are right :) I thought I tried that, but my original example was using failing assertion to write the value to the screen, so I might have mistaken the failing assertion message, and overlook the actual failure. 

The error seems to be here: 

https://github.com/OmniSharp/omnisharp-vscode/blob/7689b0d3dd615224ac890dd9bccf9fb0bdae4a64/src/features/dotnetTest.ts#L367-L368

Debugger uses custom launcher, so test platform responds with info that says how testhost should be started, instead of starting it by itself. The environment variables arrive on the request, but are not passed on.

**Comment by nohwnd** [comment]
@zhen8838  Yet again this is closed. We currently have a release pending. This will be in the next one, which is usually once a month. A possible workaround, if feasible for you, is setting the env variable in a console / terminal, and start VSCode from there. That way the whole process tree including the debugger and omnisharp server will inherit it. It is little less dynamic than providing a settings file, but should work nonetheless.

**Comment by zhen8838** [comment]
@nohwnd Thank you very much for your help! 👍

**Comment by nohwnd** [comment]
@zhen8838 I had hard time not including it in the release, so I just included it. The artifacts are now being uploaded. And there should be new version available soon, with this fix, 1.24.2.
